src/server.c

```diff
@@ -1672,8 +1672,27 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         if (listLength(server.clients_waiting_acks) && prev_fsynced_reploff != server.fsynced_reploff) dont_sleep = 1;
     }
 
-    /* Handle writes with pending output buffers. */
-    handleClientsWithPendingWritesUsingThreads();
+    /* Handle writes with pending output buffers.
+     * In the first phase, we will try to leverage io_uring to handle fsync asynchronously,
+     * just skip the client that has pending writes. */
+    handleClientsWithPendingWritesUsingThreads(1);
```

Comment:
Shouldn't we write to all clients if we are not using io_uring?
```suggestion
    handleClientsWithPendingWritesUsingThreads(canFsyncUsingIOUring());
```